gmt_machine/security/security_view.xml

```diff
@@ -0,0 +1,49 @@
+<odoo>
+    <data>
+
+        <!-- Access Group Category -->
+        <record id="post_access_categ" model="ir.module.category">
+            <field name="name">Post</field>
+        </record>
+
+        <!-- Group of Machine User -->
+        <record id="group_machine_user" model="res.groups">
+            <field name="name">Machine User</field>
+            <field name="category_id" ref="post_access_categ"/>
+        </record>
+
+        <!-- Group of Machine Supervisor -->
+        <record id="group_machine_supervisor" model="res.groups">
+            <field name="name">Machine Supervisor</field>
+            <field name="category_id" ref="post_access_categ"/>
+            <field name="implied_ids" eval="[(4, ref('group_machine_user'))]"/>
+        </record>
+
+        <!-- Group of Unit Manager -->
+        <record id="group_unit_manager" model="res.groups">
+            <field name="name">Unit Manager</field>
+            <field name="category_id" ref="post_access_categ"/>
+            <field name="implied_ids" eval="[(4, ref('group_machine_supervisor'))]"/>
+            <field name="users" eval="[(4, ref('base.user_admin'))]"/>
+        </record>
+
+        <!-- Record Rules -->
+
+        <!-- Machine access for Machine Supervisor -->
+        <record id="machine_supervisor_product_template_access_rule" model="ir.rule">
+            <field name="name">Machine:Machine Supervisor</field>
+            <field name="model_id" ref="model_product_template"/>
+            <field name="groups" eval="[(4, ref('group_machine_supervisor'))]"/>
+            <field name="domain_force">[('machine_user_id.partner_id.id','=',user.partner_id.id)]</field>
```

Comment:
This should be machine user.
Machine supervisor is for multiple machines.

In that logic, we may need to add field as Machine users (M2M users),Machine Supervisor as m2o.
Makes sense?

Because one machine can b operated by many, at various shifts.